plugin/metrics/prometheus/metricsstore/dbmodel/to_domain.go

```diff
@@ -23,37 +23,54 @@ import (
 	"github.com/jaegertracing/jaeger/proto-gen/api_v2/metrics"
 )
 
+// Translator translates Prometheus's metrics model to Jaeger's.
+type Translator struct {
+	labelMap map[string]string
+}
+
+// New returns a new Translator.
+func New(spanNameLabel string) Translator {
+	return Translator{
+		// "operator" is the label name that Jaeger UI expects.
+		labelMap: map[string]string{spanNameLabel: "operation"},
+	}
+}
+
 // ToDomainMetricsFamily converts Prometheus' representation of metrics query results to Jaeger's.
-func ToDomainMetricsFamily(name, description string, mv model.Value) (*metrics.MetricFamily, error) {
+func (d Translator) ToDomainMetricsFamily(name, description string, mv model.Value) (*metrics.MetricFamily, error) {
 	if mv.Type() != model.ValMatrix {
 		return &metrics.MetricFamily{}, fmt.Errorf("unexpected metrics ValueType: %s", mv.Type())
 	}
 	return &metrics.MetricFamily{
 		Name:    name,
 		Type:    metrics.MetricType_GAUGE,
 		Help:    description,
-		Metrics: toDomainMetrics(mv.(model.Matrix)),
+		Metrics: d.toDomainMetrics(mv.(model.Matrix)),
 	}, nil
 }
 
 // toDomainMetrics converts Prometheus' representation of metrics to Jaeger's.
-func toDomainMetrics(matrix model.Matrix) []*metrics.Metric {
+func (d Translator) toDomainMetrics(matrix model.Matrix) []*metrics.Metric {
 	ms := make([]*metrics.Metric, matrix.Len())
 	for i, ss := range matrix {
 		ms[i] = &metrics.Metric{
-			Labels:       toDomainLabels(ss.Metric),
+			Labels:       d.toDomainLabels(ss.Metric),
 			MetricPoints: toDomainMetricPoints(ss.Values),
 		}
 	}
 	return ms
 }
 
 // toDomainLabels converts Prometheus' representation of metric labels to Jaeger's.
-func toDomainLabels(promLabels model.Metric) []*metrics.Label {
+func (d Translator) toDomainLabels(promLabels model.Metric) []*metrics.Label {
 	labels := make([]*metrics.Label, len(promLabels))
 	j := 0
 	for k, v := range promLabels {
-		labels[j] = &metrics.Label{Name: string(k), Value: string(v)}
+		labelName := string(k)
+		if newLabel, ok := d.labelMap[labelName]; ok {
```

Comment:
This is needed because the spanmetrics connector renamed `operation` to `span_name`. I made it a method of a `Translator` struct because it was a little tedious (and awkward) to pass the map through multiple levels of functions.